### PR TITLE
Create SECURITY.md

### PR DESCRIPTION
**Pull request by ricab**
Add a SECURITY.md file, documenting Multipass's security policy, along with instructions for reporting vulnerabilities and keeping up to date with supported releases.

MULTI-1419

### CODE REVIEW

File: SECURITY.md [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+2. Click "Report a Vulnerability."
+3. Provide detailed information about the vulnerability, including steps to reproduce, affected versions, and potential impact.
+
+## Response to vulnerabilities
+
+Vulnerabilities are classified by risk, which factors in impact and likelihood. 
+The Multipass project will prioritize responding to all High and Critical severity vulnerabilities.
+
+When we receive an issue, we will work with the reporter to determine how best to proceed. 
+After a fix is available to a confirmed vulnerability, we will also coordinate disclosure with releasing to the various platforms.
```
Comment (Sploder12) [inline SECURITY.md:35]: This sentence is a bit confusing to me. Should the `with` be an `and`?

**Review by Sploder12 — COMMENTED** [review]
Looks good! Just some nitpicking from me.

File: SECURITY.md [hunk 2]
```diff
@@ -0,0 +1,70 @@
+# Security Policy
+
+
+## What is a vulnerability?
+
+All vulnerabilities are bugs, but not every bug is a vulnerability. Vulnerabilities compromise one or more of:
+
+- Confidentiality (personal or corporate confidential data).
+- Integrity (trustworthiness and correctness).
+- Availability (uptime and service).
```
[the remark this answers is not in the document]
Comment (ricab) [reply SECURITY.md:10]: Good point. I stole this from somewhere else and didn't even notice that.

File: SECURITY.md (hunk 1 above)
Comment (ricab) [reply SECURITY.md:35]: The intention was that the processes of disclosing and releasing had to be coordinated with each other. But I suppose it's not achieving that anyway, so I'll replace as you suggest.

**Review by jibel — DISMISSED** [review]
LGTM. Thanks!

File: SECURITY.md [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+After a fix is available to a confirmed vulnerability, we will also coordinate disclosing and releasing to the various platforms.
+
+## Supported version
+
+Multipass is released as a snap on Linux, an MSI package on Windows, and an installer package on macOS. 
+In each of these platforms, we support the latest stable version of Multipass.
+Multipass notifies you of newer releases on all platforms.
+On Linux, updates are handled automatically by the snap machinery.
+
+Please ensure you are using the latest version, to benefit from the latest patches.
```
Comment (giuliazanchi) [inline SECURITY.md:44]:
```suggestion
Please ensure you are using the latest version to benefit from the latest patches.
```

File: SECURITY.md [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+
+### Release of security fixes
+
+Security updates are distributed with a new release, which becomes the new supported version.
+This can be either a bug-fix, minor, or major release, depending on what other modifications it includes.
+
+The urgency of the fixes included - security and otherwise - determines the urgency of the release.
+We are committed to fixing high-risk security issues and releasing them as quickly as possible.
+
+In addition, the candidate channel of the Multipass snap provides frequent rebuilds of the stable channel with updated dependencies.
```
Comment (giuliazanchi) [inline SECURITY.md:54]: Not sure if this is appropriate, please review:

```suggestion
Additionally, the `candidate` channel of the Multipass snap provides frequent rebuilds of the `stable` channel with updated dependencies.
```

File: SECURITY.md [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+These builds are produced from the same Multipass code, but with up-to-date deb packages.
```
Comment (giuliazanchi) [inline SECURITY.md:55]:
```suggestion
These builds are produced from the same Multipass code, but with up-to-date `.deb` packages.
```

File: SECURITY.md [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+Candidate snaps are later promoted to the stable channel from time to time.
```
Comment (giuliazanchi) [inline SECURITY.md:56]:
```suggestion
Candidate snaps are promoted to the stable channel from time to time.
```

File: SECURITY.md [hunk 7]
```diff
@@ -0,0 +1,70 @@
...
+
+### Finding versions
+
+Multipass can tell you its version with:
```
Comment (giuliazanchi) [inline SECURITY.md:60]:
```suggestion
You can find out which version of Multipass is installed on your system with the command:
```

File: SECURITY.md [hunk 8]
```diff
@@ -0,0 +1,70 @@
...
+
+```
+multipass version
+```
+
+The snap command provides information on exact revisions and build dates:
```
Comment (giuliazanchi) [inline SECURITY.md:66]:
```suggestion
The `snap` command provides information on exact revisions and build dates:
```

File: SECURITY.md [hunk 9]
```diff
@@ -0,0 +1,70 @@
...
+
+- Confidentiality (personal and corporate confidential data).
+- Integrity (trustworthiness and correctness).
+- Availability (uptime and service).
+
+If you discover a security vulnerability within Multipass, we encourage responsible disclosure. 
+If you're not sure whether you found a vulnerability, a bug, or something else, please use the process below for reporting a vulnerability. 
+We will then assess and triage accordingly.
+
+## Reporting a Vulnerability
```
Comment (giuliazanchi) [inline SECURITY.md:16]:
```suggestion
## Reporting vulnerabilities
```

File: SECURITY.md [hunk 10]
```diff
@@ -0,0 +1,70 @@
...
+
+## Response to vulnerabilities
+
+Vulnerabilities are classified by risk, which factors in impact and likelihood. 
+The Multipass project will prioritize responding to all High and Critical severity vulnerabilities.
+
+When we receive an issue, we will work with the reporter to determine how best to proceed. 
+After a fix is available to a confirmed vulnerability, we will also coordinate disclosing and releasing to the various platforms.
+
+## Supported version
```
Comment (giuliazanchi) [inline SECURITY.md:37]:
```suggestion
## Supported versions
```

File: SECURITY.md [hunk 11]
```diff
@@ -0,0 +1,70 @@
...
+Here is the link to open a [new security adivory](https://github.com/canonical/multipass/security/advisories/new). 
+For detailed instructions, please review the documentation on [privately reporting a security vulnerability](https://docs.github.com/en/code-security/security-advisories/guidance-on-reporting-and-writing-information-about-vulnerabilities/privately-reporting-a-security-vulnerability).
+
+### Steps to Report a Vulnerability
+
+1. Go to the [Security Advisories Page](https://github.com/canonical/multipass/security/advisories) of the `multipass` repository.
+2. Click "Report a Vulnerability."
+3. Provide detailed information about the vulnerability, including steps to reproduce, affected versions, and potential impact.
+
+## Response to vulnerabilities
```
Comment (giuliazanchi) [inline SECURITY.md:29]:
```suggestion
## Our response process
```

File: SECURITY.md [hunk 12]
```diff
@@ -0,0 +1,70 @@
...
+This can be either a bug-fix, minor, or major release, depending on what other modifications it includes.
+
+The urgency of the fixes included - security and otherwise - determines the urgency of the release.
+We are committed to fixing high-risk security issues and releasing them as quickly as possible.
+
+In addition, the candidate channel of the Multipass snap provides frequent rebuilds of the stable channel with updated dependencies.
+These builds are produced from the same Multipass code, but with up-to-date deb packages.
+Candidate snaps are later promoted to the stable channel from time to time.
+
+### Finding versions
```
Comment (giuliazanchi) [inline SECURITY.md:58]:
```suggestion
### How to check your version
```

File: SECURITY.md [hunk 13]
```diff
@@ -0,0 +1,70 @@
...
+- Integrity (trustworthiness and correctness).
+- Availability (uptime and service).
+
+If you discover a security vulnerability within Multipass, we encourage responsible disclosure. 
+If you're not sure whether you found a vulnerability, a bug, or something else, please use the process below for reporting a vulnerability. 
+We will then assess and triage accordingly.
+
+## Reporting a Vulnerability
+
+Multipass accepts private reports of security vulnerabilities made through
```
Comment (giuliazanchi) [inline SECURITY.md:18]:
```suggestion
Multipass accepts private reports of security vulnerabilities submitted through 
```

File: SECURITY.md [hunk 14]
```diff
@@ -0,0 +1,70 @@
...
+Multipass accepts private reports of security vulnerabilities made through 
+[GitHub Security Advisories](https://docs.github.com/en/code-security/security-advisories/working-with-repository-security-advisories/about-repository-security-advisories). 
+Here is the link to open a [new security adivory](https://github.com/canonical/multipass/security/advisories/new). 
+For detailed instructions, please review the documentation on [privately reporting a security vulnerability](https://docs.github.com/en/code-security/security-advisories/guidance-on-reporting-and-writing-information-about-vulnerabilities/privately-reporting-a-security-vulnerability).
+
+### Steps to Report a Vulnerability
+
+1. Go to the [Security Advisories Page](https://github.com/canonical/multipass/security/advisories) of the `multipass` repository.
+2. Click "Report a Vulnerability."
+3. Provide detailed information about the vulnerability, including steps to reproduce, affected versions, and potential impact.
```
Comment (giuliazanchi) [inline SECURITY.md:27]:
```suggestion
3. Provide detailed information about the vulnerability, including steps to reproduce, affected versions, and potential impact.
4. Click "Submit report".
```

**Comment by giuliazanchi** [comment]
Hi @ricab , I've submitted my review. There are some suggestions based on our style guide, some suggested rewordings and small nitpicks.

File: SECURITY.md [hunk 15]
```diff
@@ -0,0 +1,70 @@
...
+If you discover a security vulnerability within Multipass, we encourage responsible disclosure. 
+If you're not sure whether you found a vulnerability, a bug, or something else, please use the process below for reporting a vulnerability. 
+We will then assess and triage accordingly.
+
+## Reporting a Vulnerability
+
+Multipass accepts private reports of security vulnerabilities made through 
+[GitHub Security Advisories](https://docs.github.com/en/code-security/security-advisories/working-with-repository-security-advisories/about-repository-security-advisories). 
+Here is the link to open a [new security adivory](https://github.com/canonical/multipass/security/advisories/new). 
+For detailed instructions, please review the documentation on [privately reporting a security vulnerability](https://docs.github.com/en/code-security/security-advisories/guidance-on-reporting-and-writing-information-about-vulnerabilities/privately-reporting-a-security-vulnerability).
```
[the remark this answers is not in the document]
Comment (ricab) [reply SECURITY.md:21]: I went with "GitHub's". WDYT?

**Comment by ricab** [comment]
Thanks for the review @giuliazanchi! I accepted most of your suggestions, so it's ready for another pass.

File: SECURITY.md (hunk 15 above)
[the remark this answers is not in the document]
Comment (giuliazanchi) [reply SECURITY.md]: @ricab Here I suggested "GitHub documentation" as in "Multipass documentation" or "Python documentation", meaning that the documentation is about GitHub and its features, not in the sense that the documentation is owned by GitHub. But it's a subtlety, so I guess it's fine either way :)

**Review by giuliazanchi — APPROVED** [review]
Thanks Ricardo, it looks OK to me, plus that comment of "GitHub docs" vs "GitHub's docs" that I shared elsewhere. I'll leave that choice to you :)

File: SECURITY.md (hunk 15 above)
[the remark this answers is not in the document]
Comment (ricab) [reply SECURITY.md]: OK, I will apply your suggestion on this one too.

**Comment by ricab** [comment]
@giuliazanchi and @Sploder12, are you good with this now? Any objections to merging this manually?

**Comment by giuliazanchi** [comment]
Looks good to me, @ricab

**Pull request merged**